x-pack/plugins/ml/server/models/data_frame_analytics/models_provider.ts

```diff
@@ -105,7 +101,12 @@ export function modelsProvider(
           const nodeFields = pick(node, NODE_FIELDS) as RequiredNodeFields;
 
           const allocatedModels = (
-            deploymentStats.deployment_stats as TrainedModelDeploymentStatsResponse[]
+            trainedModelStats
+              .map((v) => {
+                // @ts-ignore new prop
+                return v.deployment_stats;
```

Comment:
nit: should this ts-ignore be here?

Reply:
Yes, we need it, because elasticsearch client hasn't been updated yet

Reply:
super duper nit: If it's due to outdated types from elasticsearch client, we should use ts-expect-error so that once the types are updated, it's easier to spot and remove this line.